ddev/src/ddev/utils/hatch.py

```diff
@@ -0,0 +1,122 @@
+# (C) Datadog, Inc. 2025-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Protocol, overload
+
+from pydantic import BaseModel, Field, RootModel, model_validator
+
+if TYPE_CHECKING:
+    from collections.abc import Collection
+    from typing import Any, Literal
+
+    from ddev.integration.core import Integration
+
+    from .platform import Platform
+
+
+class HatchCommandError(Exception):
+    pass
+
+
+class Environment(BaseModel):
+    """Represents a single environment configuration."""
+
+    name: str
+    type: str
+    dependencies: list[str] = Field(default_factory=list)
+    test_env: bool = Field(alias='test-env')
+    e2e_env: bool = Field(alias='e2e-env')
+    benchmark_env: bool = Field(alias='benchmark-env')
+    latest_env: bool = Field(alias='latest-env')
+    python: str | None = None
+    scripts: dict[str, list[str]] = Field(default_factory=dict)
+    platforms: list[str] = Field(default_factory=list)
+    pre_install_commands: list[str] = Field(default_factory=list, alias='pre-install-commands')
+    post_install_commands: list[str] = Field(default_factory=list, alias='post-install-commands')
+    skip_install: bool = Field(False, alias='skip-install')
+
+
+class HatchEnvironmentConfiguration(RootModel[list[Environment]]):
+    """
+    A root model that parses the top-level dictionary returned by the `hatch env show --json` command
+    into a list of Environment models.
+    """
```

Comment:
```suggestion
    """
    root: list[Environment]

    def __iter__(self):
        return iter(self.root)

    def __getitem__(self, item):
        return self.root[item]
```
nit: this would make iterating over the list of environments cleaner without having to access the `.root` attribute.